### PR TITLE
Mvp

### PR DESCRIPTION
**Pull request by Dorothea87**

### CODE REVIEW

File: src/main/scala/Cafe/MenuItem.scala [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package Cafe
 
+
 case class MenuItem(name: String, price: Double, foodType: FoodType)
```
Comment (nimo-mohamed) [inline src/main/scala/Cafe/MenuItem.scala]: suggestion: 

- Right now, an order can be placed even if stockCount == 0.

suggested fix: case class MenuItem(name: String, price: Double, stockCount: Int, foodType: FoodType) {
  def isAvailable: Boolean = stockCount > 0
}

File: src/main/scala/Cafe/Order.scala [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package Cafe
 
+
 case class Order(items: List[MenuItem]) {
```
Comment (nimo-mohamed) [inline src/main/scala/Cafe/Order.scala]: suggestions:

- No service charge logic yet.
- It calculates the total price without applying the service charge.

suggested fix: case class Order(items: List[MenuItem]) {
  def orderTotal: Double = items.map(_.price).sum

  def serviceCharge: Double = {
    items match {
      case i if i.exists(_.foodType == Drinks) => 0.0
      case i if i.exists(_.foodType == ColdFood) => orderTotal * 0.10
case.............ADD THE REST
    }
  }

  def totalWithServiceCharge: Double = orderTotal + serviceCharge
}


- This adds automatic service charge calculation based on the menu item types.
- Uses pattern matching instead of if-else.

File: src/test/scala/CafeLogicSpec.scala [hunk 3]
```diff
@@ -3,6 +3,8 @@ import Cafe.{MenuList, Order}
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 
+//haven't tested the createOrder Method yet
+
 class CafeLogicSpec extends AnyWordSpec with Matchers {
```
Comment (nimo-mohamed) [inline src/test/scala/CafeLogicSpec.scala:8]: suggested things to test here later:

- you can write a test for **createdBill** method

File: src/test/scala/OrderSpec.scala [hunk 4]
```diff
@@ -2,6 +2,7 @@ import Cafe.{MenuList, Order}
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 
+
 class OrderSpec extends AnyWordSpec with Matchers {
```
Comment (nimo-mohamed) [inline src/test/scala/OrderSpec.scala:6]: suggested tests for this file: test to verify **service charge calculations**

here is some edge cases to consider: 

No service charge for drinks.
10% for cold food.
20% for hot food.
25% for premium items.

File: src/main/scala/Cafe/CafeLogic.scala [hunk 5]
```diff
@@ -35,9 +35,11 @@ object CafeLogic extends App {
 
 
   //this is my to do list (sorry, I'm slow)
```
Comment (nimo-mohamed) [inline src/main/scala/Cafe/CafeLogic.scala]: OMGGG this was folded up and I didn't see all the code above😫

I love your todo list and I can see where your heading with this💖

I love how well structured the project is 
good use of separate files for concerns 
Excellent usage of case class for MenuItem, Order, etc.
tests are structured correctly 👍
great test setup "Nimo will be copying this 😊"
Tests are structured using AnyWordSpec with Matchers, which is a good testing pattern.👍
good use of case classes 👍
you insured there was type safety by defining FoodType & MenuCategory as sealed trait👍

File: src/test/scala/CafeLogicSpec.scala [hunk 6]
```diff
@@ -3,6 +3,8 @@ import Cafe.{MenuList, Order}
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 
+//haven't tested the createOrder Method yet
+
```
Comment (nimo-mohamed) [inline src/test/scala/CafeLogicSpec.scala]: Ensure total price is formatted as £X.XX (two decimal places).  if formatting changes slightly (e.g., an extra space or different newline style), this test might fail even though the logic is correct.
Currency Formatting – £7.9 should be £7.90.

File: src/test/scala/MenuListSpec.scala [hunk 7]
```diff
@@ -2,6 +2,7 @@ import Cafe.{MenuItem, MenuList, PremiumItem}
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 
+
 class MenuListSpec extends AnyWordSpec with Matchers {
```
Comment (nimo-mohamed) [inline src/test/scala/MenuListSpec.scala:8]: - The **addPremiumItem** method should check if an item **already exists** before adding it.
- Right now, it's not testing for **duplicate handling**.
